metals/src/main/scala/scala/meta/internal/metals/BuildTargetInfo.scala

```diff
@@ -0,0 +1,306 @@
+          info += ""
+          info += "Classes Directory"
+          info += s"  ${java.classesDirectory}"
+        })
+      else {
+        detail.java.foreach(java => {
+          info += ""
+          info += "Java Classes Directory"
+          info += s"  ${java.classesDirectory}"
+        })
+        detail.scala.foreach(scala => {
+          info += ""
+          info += "Scala Classes Directory"
+          info += s"  ${scala.classesDirectory}"
+        })
+      }
+      if (detail.sameJavaScalaClasspaths)
+        detail.java.foreach(java => {
+          info += ""
+          info += "Classpath"
+          addClassPath(info, java.classPath)
+        })
+      else {
+        detail.java.foreach(java => {
+          info += ""
+          info += "Java Classpath"
+          addClassPath(info, java.classPath)
+        })
+        detail.scala.foreach(scala => {
+          info += ""
+          info += "Scala Classpath"
+          addClassPath(info, scala.classPath)
+        })
+      }
+      info += ""
+      info.mkString(System.lineSeparator())
+    })
+  }
+
+  private def addClassPath(
+      info: ListBuffer[String],
+      classPath: List[Path]
+  ): Unit = {
+    def convertToName(path: Path): Path = {
+      if (path.toFile.isFile)
+        path.getFileName()
+      else
+        path
+    }
+    if (classPath.nonEmpty) {
+      val maxFilenameSize =
+        classPath.map(convertToName(_).toString.length()).max + 5
+      classPath.foreach(f => {
+        val filename = convertToName(f).toString()
+        val padding = " " * (maxFilenameSize - filename.size)
+        val status = if (f.toFile.exists) {
+          if (f.toFile().isDirectory())
+            "         "
+          else {
+            val sourceJarName = filename.replace(".jar", "-sources.jar")
+            val hasSource = buildTargets
+              .sourceJarFile(sourceJarName)
+              .exists(_.toFile.exists())
+            if (hasSource)
+              "         "
+            else
+              "NO SOURCE"
+          }
+        } else " MISSING "
+        val fullName = if (f.toFile.isFile) s" $f" else ""
+        info += s"  $filename$padding $status$fullName"
+      })
+    } else
+      info += "  NONE"
+  }
+
+  private case class TargetDetail(
+      common: CommonTargetDetail,
+      java: Option[JavaTargetDetail],
+      scala: Option[ScalaTargetDetail]
+  ) {
+    def sameJavaScalaClasspaths: Boolean =
+      java.nonEmpty && scala.nonEmpty && java.get.classPath == scala.get.classPath
+    def sameJavaScalaClassesDir: Boolean =
+      java.nonEmpty && scala.nonEmpty && java.get.classesDirectory == scala.get.classesDirectory
+  }
+  private case class CommonTargetDetail(
+      id: BuildTargetIdentifier,
+      displayName: String,
+      baseDirectory: String,
+      tags: List[String],
+      languageIds: List[String],
+      dependencies: List[String],
+      dependentTargets: List[String],
+      capabilities: List[(String, Boolean)],
```

Comment:
It'd be nicer if this could be a `List[BuildTargetCapability]`. Then we could also avoid the `._2` shenanigans up above.

Reply:
Actually thinking about this, there seems to be a lot of redundancy in here. Are all of these classes actually needed. For example we could also just use `BuildTargetCapabilities` from BSP for this couldn't we? Same with almost all of this stuff in here. This is very close to the actual `BuildTarget` interface.